Services/RefundService.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+namespace MollieShopware\Services;
+
+use Doctrine\ORM\EntityManager;
+use Exception;
+use Mollie\Api\Resources\Order as MollieOrder;
+use Mollie\Api\Resources\Payment as MolliePayment;
+use Mollie\Api\Resources\OrderLine;
+use Mollie\Api\Resources\Refund;
+use MollieShopware\Models\OrderLines;
+use MollieShopware\Models\OrderLinesRepository;
+use MollieShopware\Models\Transaction;
+use MollieShopware\Traits\MollieApiClientTrait;
+use Shopware\Models\Order\Detail;
+use Shopware\Models\Order\Order;
+use Shopware\Models\Order\Status;
+
+class RefundService implements RefundInterface
+{
+    use MollieApiClientTrait;
+
+    /**
+     * @var EntityManager
+     */
+    private $modelManager;
+
+    public function __construct(EntityManager $modelManager)
+    {
+        $this->modelManager = $modelManager;
+    }
+
+    public function refundOrderAmount(Order $order, Transaction $transaction, float $customAmount = null)
+    {
+        $mollieClient = $this->getMollieApi($order->getShop()->getId());
```

Comment:
in the code, there is a MollieShopSwitcher which was introduced as some small helper because of a problem with sub shops....
could you double check that we use the correct plugin configuration (api keys, etc...) of that particular order.
if that order is from subshop B, it could be that subshop B has a different API key...just to make sure its working thank you